### PR TITLE
A patch for 199 showing kwargs only decorator

### PR DESCRIPTION
**Pull request by dhakim87**
Implementation restricting decorator to kwargs only functions.  It cleans up the decorator, but I feel it significantly increases the barrier to entry of client_impl for devs who are new to python (like me).  It's here as an option, but I prefer the decorator_util bound mappings version

### CODE REVIEW

**Comment by wasade** [comment]
The more I look at this, the more this looks like the safest long term approach

File: microsetta_private_api/util/decorator_util.py [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
 
+def has_non_keyword_arguments(func):
+    sig = inspect.signature(func)
+    params = sig.parameters
+    for p in params:
+        if params[p].kind != inspect.Parameter.KEYWORD_ONLY:
+            return True
+
+
 def build_param_map(func, param_names):
```
Comment (wasade) [inline microsetta_private_api/util/decorator_util.py:12]: This is no longer used, right?

Comment (wasade) [reply microsetta_private_api/util/decorator_util.py:12]: ...same with `bind_param_map`

Comment (dhakim87) [reply microsetta_private_api/util/decorator_util.py]: Removed.

**Pull request merged**